File: 151051/Krupovich/310 & 320/dc_rest/src/main/java/ru/bsuir/mapper/LabelMapper.java
package ru.bsuir.mapper;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import ru.bsuir.dto.request.LabelRequestTo;
import ru.bsuir.dto.response.LabelResponseTo;
import ru.bsuir.entity.Label;

@Mapper(componentModel = "spring")
public interface LabelMapper {

    @Mapping(target = "id", ignore = true)

    Label toEntity(LabelRequestTo dto);
    LabelResponseTo toDTO(Label entity);
}

File: 251003/Pankratiev/rest/src/main/kotlin/bsuir/dc/rest/mapper/LabelMapper.kt
package bsuir.dc.rest.mapper

import bsuir.dc.rest.dto.from.LabelRequestTo
import bsuir.dc.rest.dto.to.LabelResponseTo
import bsuir.dc.rest.entity.Label

fun LabelRequestTo.toEntity() = Label(
    id = this.id,
    name = this.name
)

fun Label.toResponse() = LabelResponseTo(
    id = this.id,
    name = this.name
)
